### PR TITLE
Add Lambda support for Skygear Data Type

### PR DESCRIPTION
**Pull request by cheungpat**
connects #100

### CODE REVIEW

**Comment by cheungpat** [comment]
@carmenlau Updated, I think this pull request is ready for merging as we have reached consensus in product meeting. Let me know if I have missed anything : )

File: features/100-lambda-params/api.md [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+Since Skygear Record was not considered a data type, the protocol does
+not have support for specifying a Record object. To support using Record object
+in parameters and return value, a new `$type=record` will be implemented.
+
+### Breaking Changes
+
+When upgrading cloud function runtime (i.e. python or node), the function
+arguments will replace `$type` JSON objects with native Skygear Data Type.
+If the developer sends `$type` JSON objects from SDK, they will need to update
```
Comment (chpapa) [inline features/100-lambda-params/api.md]: Just want to confirm what I heard from product meeting, we are pretty sure right now most users don't do it? (Maybe a quick survey in office?)

**Comment by chpapa** [comment]
Great work and overall looks good. 

May I know if there are any use cases (or considered?) for Cloud Functions that return an asset object? @cheungpat @carmenlau

**Comment by limouren** [comment]
For `$type=record`, would it cover record's `$transient` field?

File: features/100-lambda-params/api.md (hunk 1 above)
Comment (carmenlau) [reply features/100-lambda-params/api.md]: Did a quick survey, no one in the office will use `$type` in their lambda response, it is safe for us to do it.

**Comment by carmenlau** [comment]
### For asset return
@chpapa 
Discussed with @cheungpat, also asked some colleagues in office, they sometime will return asset from the lambda. To facilitate the user to returning lambda, if user return record with asset or return asset directly. We will signed the asset, and user will be able to give the signed asset in lambda result.

### For $transient
@limouren 
If the returned record in lambda response with `$transient`, we will still return it to the SDK and user will be able to access the `$transient` in record. But if user sending request to lambda with record that contain `$transient`, we will remove and ignore it just like we did record save.

Let me know if there is any further questions or comments? Otherwise I will merge this spec and start the development! Thanks :)

**Pull request merged**